### PR TITLE
chore(website): Auto generate sources/destinations sidebar/tables

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

This is a follow up to https://github.com/cloudquery/cloudquery/issues/7004.
Nextra added support for using JS in in `_meta.json` files so we can auto generate the sidebar and tables.

See examples of us forgetting to do those:
1. https://github.com/cloudquery/cloudquery/pull/7112
2. https://github.com/cloudquery/cloudquery/pull/7014
3. On https://www.cloudquery.io/docs/plugins/destinations/overview
image

This will also ensure the plugins are sorted by name and we won't need https://github.com/cloudquery/cloudquery/pull/7223.
Example of unsorted entries:
image

It does so by adding `title:` and `stage:` frontmatter entries to each plugin's overview instead of the `_meta.json` entries. This should also save us `git` conflicts.

### CODE REVIEW

File: website/pages/docs/plugins/destinations/_meta.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const { getPluginsData } = require("../../../../utils/plugins");
+
+const getMeta = () => {
+  const pluginsData = getPluginsData("destinations");
+  const asMeta = Object.entries(pluginsData).map(([destination, { name }]) => [destination, name]);
+  const meta = {
+    overview: "Overview",
```
Comment (erezrokah) [inline website/pages/docs/plugins/destinations/_meta.js:7]: Put `Overview` first

File: website/pages/docs/plugins/destinations/overview.mdx [hunk 2]
```diff
@@ -3,32 +3,28 @@ title: Destinations
 ---
 
 import { PluginsTable } from "../../../../components/PluginsTable";
+import { getPluginsData } from '../../../../utils/plugins'
+import { useSSG } from 'nextra/ssg'
+
+{/* <!-- vale off --> */}
+export const getStaticProps = () => {
```
Comment (erezrokah) [inline website/pages/docs/plugins/destinations/overview.mdx:10]: We need `getStaticProps` as `getPluginsData` uses `fs` so we need to tell Next.js to run it on "Server" context. This should run only once during build as we don't revalidate it

File: website/pages/docs/plugins/destinations/_meta.json [hunk 3]
```diff
@@ -1,14 +0,0 @@
...
-  "bigquery": "BigQuery",
-  "csv": "CSV (Deprecated)",
-  "file": "File",
-  "gcs": "GCS",
-  "kafka": "Kafka",
-  "mongodb": "MongoDB",
-  "mssql": "Microsoft SQL Server",
-  "postgresql": "PostgreSQL",
-  "snowflake": "Snowflake",
-  "sqlite": "SQLite"
```
Comment (erezrokah) [inline website/pages/docs/plugins/destinations/_meta.json:13]: This was missing S3 and Neo4j

File: website/pages/docs/plugins/destinations/postgresql/overview.mdx [hunk 4]
```diff
@@ -1,3 +1,7 @@
+---
+name: PostgreSQL
+stage: GA
```
Comment (erezrokah) [inline website/pages/docs/plugins/destinations/postgresql/overview.mdx:3]: The default is `Preview`

File: website/pages/docs/plugins/sources/plausible/overview.mdx [hunk 5]
```diff
@@ -1,3 +1,6 @@
+---
+name: PagerDuty
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/plausible/overview.mdx:2]:
```suggestion
name: Plausible Analytics
```

File: website/pages/docs/plugins/sources/plausible/overview.mdx [hunk 6]
```diff
@@ -1,3 +1,6 @@
+---
+name: Plausible Analytics
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/plausible/overview.mdx:2]:
```suggestion
name: Plausible
```

File: website/utils/plugins.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+const fs = require("fs");
+const path = require("path");
+const matter = require("gray-matter");
```
Comment (erezrokah) [inline website/utils/plugins.js:3]: I used `gray-matter` and `title` as those are used by Nextra so we have the same logic

**Review by candiduslynx — APPROVED** [review]
Nice!

**Review by yevgenypats — APPROVED** [review]
🚀

**Comment by hermanschaaf** [comment]
Nice! 🚀

**Pull request merged**